apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -121,7 +121,9 @@ pub fn build_supergraph_api_query_graph(
     let mut override_labels_by_field = IndexMap::default();
     for (pos, node) in join_field_applications {
         let Ok(pos) = FieldDefinitionPosition::try_from(pos.clone()) else {
-            bail!("unexpected @join__field directive application position at {pos}");
+            // @join__field can also appear on input object fields, which can't be overridden, so we
+            // just skip those fields here.
+            continue;
```

Comment:
do we need to check if `@join__field` appears on other position? guessing no (as otherwise it would be invalid schema) but just double checking